packages/firestore/test/integration/util/helpers.ts

```diff
@@ -208,32 +217,19 @@ export function withTestDbsSettings(
     promises.push(ready);
   }
 
-  return Promise.all(promises).then((dbs: firestore.FirebaseFirestore[]) => {
-    const cleanup = () => {
-      return wipeDb(dbs[0]).then(() =>
-        dbs.reduce(
-          (chain, db) =>
-            chain.then(
-              db.INTERNAL.delete.bind(this, {
-                purgePersistenceWithDataLoss: true
-              })
-            ),
-          Promise.resolve()
-        )
-      );
-    };
-
-    return fn(dbs).then(
-      () => cleanup(),
-      err => {
-        // Do cleanup but propagate original error.
-        return cleanup().then(
-          () => Promise.reject(err),
-          () => Promise.reject(err)
-        );
-      }
-    );
-  });
+  const dbs = await Promise.all(promises);
+
+  try {
+    await fn(dbs);
+  } finally {
+    await wipeDb(dbs[0]);
+    for (let i = 0; i < dbs.length; i++) {
```

Comment:
Since you don't otherwise need the index, consider using a `for..of` loop instead:

```ts
for (const db of dbs) {
  await db.INTERNAL.delete();
}
```